_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,234 @@
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+input_aspects:
+- DOM Tree
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+This rule applies to `th` elements within a `table`, where the `table` is [visible](#visible) and [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+## Expectation
+
+Each target element is either the row header or the column header for data cells within the same `table`.
+
+## Assumptions
+
+- Tables are [well-formed, according to the HTML5.1 specification](https://www.w3.org/TR/html51/tabular-data.html#forming-a-table).
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [Forming relationships between data cells and header cells](https://www.w3.org/TR/html/tabular-data.html#forming-relationships-between-data-cells-and-header-cells)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+`th` has corresponding `cells` within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th>Time</th>
+	</tr>
+	<tr>
+		<td>05:41</td>
+	</tr>
+</table>
+```
+
+#### Passed Example 2
+
+`th` has corresponding `cells` within the same `table`. Here the cells span multiple columns.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th>Projects</th>
+      <th>Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `columnheader` has corresponding `cells` within the same `table`, by usage of `headers` attribute.
+
+```html
+<table>
+  <tr> 
+    <td headers="header1">
+      Projects
+    </td>
+    <th id="header1">
+      Projects
+    </th> 
+  </tr>
+</table>
+```
+
+#### Passed Example 4
+
+The `columnheader` has corresponding `cells` within the same `table`, by usage of `aria-labelledby` attribute.
+
+```html
+<table>
+	<tr>
+    <th id="hd2">Time</th>
+    <td aria-labelledby="hd2">05:44</td>
+	</tr>
+</table>
+```
+
+#### Passed example 5
+
+Each `th` element has corresponding `cells` within the same `table`, by usage of `scope` attribute.
+
+```html
+<table>
+  <caption>Hogwarts Houses stars</caption>
+  <tr>
+      <th scope="col">Name</th>
+      <th scope="col">House</th>
+  </tr>
+  <tr>
+      <th scope="row">Harry Potter</th>
+      <td>Gryffindor</td>
+  </tr>
+  <tr>
+      <th scope="row">Cedric Diggory</th>
+      <td>Hufflepuff</td>
+  </tr>
+  <tr>
+      <th scope="row">Cho Chang</th>
+      <td>Ravenclaw</td>
+  </tr>
+    <tr>
+      <th scope="row">Severus Snape</th>
+      <td>Slytherin</td>
+  </tr>
+</table>
+```
+
```

Comment:
I think it would be nice to have an example with `th` both as column and row headers but without explicit roles or the use of the `scope` attribute. Something like:
```html
<table>
    <caption>Hogwarts Houses stars</caption>
    <tr>
        <th>Name</th>
        <th>House</th>
    </tr>
    <tr>
        <th>Harry Potter</th>
        <td>Gryffindor</td>
    </tr>
    <tr>
        <th>Cedric Diggory</th>
        <td>Hufflepuff</td>
    </tr>
    <tr>
        <th>Cho Chang</th>
        <td>Ravenclaw</td>
    </tr>
    <tr>
        <th>Severus Snape</th>
        <td>Slytherin</td>
    </tr>
</table>
```